Zend/Zend.m4

```diff
@@ -297,19 +297,35 @@ fi
 AC_MSG_CHECKING(whether to enable zend signal handling)
 AC_MSG_RESULT($ZEND_SIGNALS)
 
+
+dnl Enable Zend Max Execution Timers by default on macOS
+AS_CASE(["$host_alias"], [*darwin*], [ZEND_MAX_EXECUTION_TIMERS="yes"], [ZEND_MAX_EXECUTION_TIMERS=$ZEND_ZTS])
+
 dnl Don't enable Zend Max Execution Timers by default until PHP 8.3 to not break the ABI
 AC_ARG_ENABLE([zend-max-execution-timers],
   [AS_HELP_STRING([--enable-zend-max-execution-timers],
     [whether to enable zend max execution timers])],
     [ZEND_MAX_EXECUTION_TIMERS=$enableval],
-    [ZEND_MAX_EXECUTION_TIMERS=$ZEND_ZTS])
-
-AS_CASE(["$host_alias"], [*linux*|*freebsd*], [], [ZEND_MAX_EXECUTION_TIMERS='no'])
-
-PHP_CHECK_FUNC(timer_create, rt)
-if test "$ac_cv_func_timer_create" != "yes"; then
-  ZEND_MAX_EXECUTION_TIMERS='no'
-fi
+    [])
+
+AS_CASE(
+  ["$host_alias"],
+  [*linux*|*freebsd*], [
+    PHP_CHECK_FUNC(timer_create, rt)
+    if test "$ac_cv_func_timer_create" != "yes"; then
+      ZEND_MAX_EXECUTION_TIMERS='no'
+    fi
+  ],
+  [*darwin*], [],
+  [ZEND_MAX_EXECUTION_TIMERS='no']
+)
+
+AS_CASE(["$host_alias"], [*darwin*], [], [
+  PHP_CHECK_FUNC(timer_create, rt)
+  if test "$ac_cv_func_timer_create" != "yes"; then
+    ZEND_MAX_EXECUTION_TIMERS='no'
+  fi
+])
```

Comment:
I'm not sure my understanding of this file is right, but isn't the last AC_CASE redundant with the two other ones?

Reply:
Not really:

- this is enabled by default only if: (Apple && (ZTS || NTS)) || ((Linux || FreeBSD) && ZTS)
- this is always disabled if: (FreeBSD or Linux) && timer_create() not available.